### ISSUE TITLE
Create catalog of existing datasets

### ISSUE THREAD

**Issue by rufuspollock** [openspending/openspending#1484]
Extract from the MetaStore (elasticsearch) a dump of all the dataset metadata. 

When doing the migration i want a full list of all datasets and their ids so that I can a) filter them to identify what to migrate b) find associated files e.g. in storage c) use this list to drive the migration

## Acceptance

* [x] Extract full dump. **DONE**: Stored in Google Drive.
* [x] Get a sample of 2-3 entries (shorn of private info) so that we get a sense (review with Rufus). – **DONE**: Reviewed. We now have a full list [in this sheet](https://docs.google.com/spreadsheets/d/1Cno0jUkfl8ozf0qjaNhuCmJEmMaL2U65sNONqVTEEnM/edit#gid=402256245).
* [x] convert metadata to spreadsheet and do filter (you may not need all fields). – **DONE**: filters were applied.
* [x] (?) remove private info and publish the catalog to the audit repo we have. – **DONE**: Private info removed (we can retrieve it all later if needed). The sheet is linked in the [README of the audit repo](https://github.com/openspending/audit-for-migration-2020/#more-on-the-openspending-database).

**Comment by sglavoie** [comment]
* :heavy_check_mark: We have a full dump from Elasticsearch (JSON).
* :heavy_check_mark: From that dump, I extracted the metadata about 3 tables, stripping away private URL, names and emails:
  * https://drive.google.com/file/d/1GZz3skMGhA9bC1BQBTQjULs6xsLJBsZQ/view?usp=sharing
  * https://drive.google.com/file/d/16YOlGbnzBtN-v1FZBvxPMrt1sQTd5r2d/view?usp=sharing
  * https://drive.google.com/file/d/1sCwmCxRSJjkgm1N5ta4aNMP2M6m8dDsQ/view?usp=sharing

**Comment by rufuspollock** [comment]
I prefer tables in google spreadsheet please with key fields. (maybe just
sample to start with).
_________________________________________________________________________
*Datopian *| https://datopian.com | Open solutions for a data driven world
*DataHub*  | https://datahub.io   | GitHub for data
*CKAN    * | http://ckan.org <https://ckan.org/>      | The world's leading
data portal solution

President - +44 7795176976 - @rufuspollock

**Comment by sglavoie** [comment]
**TL;DR**: Here is the CSV for a portion of the dump: https://docs.google.com/spreadsheets/d/1e6dDIjcdBZthEsEckvd1tOTz94DksuZf7l5vRxR3WHM/edit?usp=sharing

It contains the key fields and to the right a list of the columns as they are stored in the PostgreSQL database – those expand to the right so that we have only one row per table, making it easier to read.

There are a few more fields that could be added and would provide additional info. Here they are for future reference:

* count_of_rows
* last_update (field in epoch time) [new info, complements what we already have]
* fiscalPeriod – if available [new info, complements what we already have]
* bytes – if available

**Comment by rufuspollock** [comment]
@sglavoie as per comment in spreadsheet:

* Can i understand if we can connect this stuff with the data stored in s3. As per discussion with adam key thing here is to have list of datasets with the *identifier* that allows us to locate the folder in s3 with datapackage.json and data.
* have we stored a full dump somewhere privately in drive?

**Comment by sglavoie** [comment]
**TL;DR**: @rufuspollock, the spreadsheet now contains a column `id` which is the identifier we need for S3 and we do have a full dump in the shared Datopian Google Drive: https://drive.google.com/drive/u/1/folders/1t7FtW1mBVOCu8zrwRz0sh-TgESh03CrG

I added the following fields:

* Column `C` is now `id`, which should be the "identifier" we are looking for as it corresponds to what appears in the URL in column `E`.
* Columns `H` and `I` now contain the `fiscal_start` and `fiscal_end` dates, which could be useful.
* Column `J` contains `last_update`, which can help us see what might be more recent independently of the date at which it was uploaded.

**Comment by rufuspollock** [comment]
@sglavoie feedback on comms: the spreadsheet is not linked in your comment so the email i got did not have it and i had to come here. Then when i looked here it is not in description or your latest comment so i have to scroll through the comments to find what i think it is. => think about the person receiving my message and what they would need to be able to act on it quickly (esp if within their email client) 😄

**Comment by sglavoie** [comment]
Thank you for your feedback. I will be careful to link the necessary references by assuming that everything is first read from an email notification. :+1:

**Comment by rufuspollock** [comment]
@sglavoie what is the state here? Can we prioritize getting this complete (how much time will that take?)

**Comment by sglavoie** [comment]
**TL;DR:** @rufuspollock, if what's [in the CSV file](https://docs.google.com/spreadsheets/d/1e6dDIjcdBZthEsEckvd1tOTz94DksuZf7l5vRxR3WHM/edit?usp=sharing) (dump samples) is what we want, getting all the metadata for all the tables and publishing it to the audit repo should take me about 2 hours to finish.

The reason for this is that the full dump we got later contains a few different fields from the Elasticsearch request and comes in one huge JSON files instead of various smaller ones, so we need to modify slightly the way in which we process it. Victor initially helped me retrieve everything but it was not made in the "correct" way, which we have now [documented in the audit repo](https://github.com/openspending/audit-for-migration-2020/#how-we-retrieve-the-metadata-from-elasticsearch).

**Comment by rufuspollock** [comment]
@sglavoie i think the google spreadsheet columns look fine. The main thing is we have the datapackage.json which gives us author, title etc. Note some of these datasets seem to have "Test" in the title (e.g. the first one) so next step would be to filter for what we want to migrate.

**Comment by sglavoie** [comment]
@rufuspollock: understood! That should not change the estimate at all because the plan would be to retrieve all the metadata and apply the existing filters to the list of tables we get. :+1:

**Comment by sglavoie** [comment]
**TL;DR**: @rufuspollock, [catalog of existing OpenSpending datasets created here](https://docs.google.com/spreadsheets/d/1Cno0jUkfl8ozf0qjaNhuCmJEmMaL2U65sNONqVTEEnM/edit#gid=402256245) – filters are already applied so we have an excellent idea of what to migrate from that list.

---

How we got those results:

* We get our complete list of tables from the Postgres database as a reference [in the sheet `OS_db_tables_count_size_io`](https://docs.google.com/spreadsheets/d/1Cno0jUkfl8ozf0qjaNhuCmJEmMaL2U65sNONqVTEEnM/edit#gid=24265109);
* We apply our list of filters to know what to keep (depending on size, table name, duplicate, test data, etc.) to get a pre-selection from this original list (the filters are discussed in #1482);
* We retrieve the full dump from Elasticsearch, on which we apply similar filters to discard what is not useful. Additionally, we do not keep tables that do not have a URL associated with them (and therefore no JSON data package file to make use of);
* We output the results from the full dump in the [`full_dump`](https://docs.google.com/spreadsheets/d/1Cno0jUkfl8ozf0qjaNhuCmJEmMaL2U65sNONqVTEEnM/edit#gid=402256245) sheet here;
* We apply a formula to match all our tables in the `full_dump` sheet with those found in the sheet `OS_db_tables_count_size_io` and we keep only the ones we planned to keep in `OS_db_tables_count_size_io` (based on the "yes"/"no" result of the column `D` in the sheet `OS_db_tables_count_size_io`).

**Comment by sglavoie** [comment]
**FIXED**: Closing as all tasks were done.

* Feel free to re-open if something more might be necessary to be done as part of the initial tasks.
